src/DonorProfiles/Repositories/Donations.php

```diff
@@ -75,31 +75,25 @@ private function getDonationAggregate( $rawAggregate, $donorId ) {
 	 */
 	protected function getDonationIDs( $donorId ) {
 		global $wpdb;
-
 		$result = DB::get_results(
 			DB::prepare(
 				"
-                SELECT 
-                    donation_id as id
-                FROM 
-                    {$wpdb->give_revenue} as revenue
-                INNER JOIN 
-                    {$wpdb->posts} as posts ON revenue.donation_id = posts.ID
-                WHERE 
-                    posts.post_author = %d
-                AND 
-                    posts.post_status IN ( 'publish', 'give_subscription' )
-				",
-				$donorId
+				SELECT revenue.donation_id as id
+				FROM {$wpdb->give_revenue} as revenue
+					INNER JOIN {$wpdb->posts} as posts ON revenue.donation_id = posts.ID
+					INNER JOIN {$wpdb->prefix}give_donationmeta as donationmeta ON revenue.donation_id = donationmeta.donation_id
```

Comment:
Is it necessary to use the `give_donationmeta` table? I mean, you can get donation ids by joining the `post` table and checking the post_author, exactly like you did before.

Reply:
Because GiveWP allows donors to make donations without creating a WP account, there are instances there a donation ID cannot be reached by checking the `post_author`. The only way to be certain you are querying all donations for a given donor ID is to check against the donationmeta, where the `_give_payment_donor_id` attribute is stored. It's frustrating, and not entirely efficient, but for the moment (until a future refactor of the donor registration signup system), it is what needs to be done.

Reply:
That makes sense. Thanks for the explanation :+1:

Reply:
You explained all in the description, but I skipped it :facepalm: 
I went to the `Files changed `tab immediately without checking the description first.